src/test/kotlin/com/bardsoftware/backend/fleet/rmarkdown/PandocTest.kt

```diff
@@ -0,0 +1,55 @@
+/**
+    Copyright 2018 BarD Software s.r.o
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Author: Mikhail Shavkunov (@shavkunov)
+ */
+package com.bardsoftware.backend.fleet.rmarkdown
+
+import com.google.common.io.Files
+import com.typesafe.config.ConfigFactory
+import com.typesafe.config.ConfigValueFactory
+import junit.framework.TestCase.assertTrue
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.nio.file.Paths
+
+class PandocTest {
```

Comment:
This test is a little bit pointless because it basically tests how its own command line with its own arguments executes in a process. I suggest replacing it with a test which creates MarkdownTaskReceiver with test config and sends a message  to that instance (maybe already parsed one, not necessarily PubSubMessage). This way you will cover more real code and make sure that command line placeholders are filled with arguments properly.

Reply:
Yes, you can, if that's what you really want to test, but my point is that you basically test the code written in test. You supply comand line from test, with argument names defined in test and values defined by test, and touch only a small piece of production code which creates a process. The point of tests, however, is to covers more production code.